src/arithmetic/algebraic_expression/algebraic_expression_construction.c

```diff
@@ -44,7 +44,7 @@ static inline bool _should_populate_edge
 	QGEdge *e
 ) {
 	ASSERT(e != NULL);
-	return (_referred_entity(e->alias) || QGEdge_VariableLength(e));
```

Comment:
an edge taking part in a `named-path` e.g. `MATCH P = (a)-[:R]->(b)` should be marked as a referenced entity.
As such with the given comment on `qg_edge.h` I believe we can revert this change back to its original version.